### DISCUSSION TITLE
Error in ./ghpc deploy hpc-slurm

### DISCUSSION THREAD

**Discussion by gpkmohan** [GoogleCloudPlatform/cluster-toolkit#2755] [General]
Dear List:

I try to hreate a simple HPC from "Guide me" option from: https://cloud.google.com/hpc-toolkit/docs/quickstarts/slurm-cluster

I followed the steps:

1	gcloud iam service-accounts enable \
    --project=ambient-depth-413708 \
    648135870514-compute@developer.gserviceaccount.com

2	gcloud projects \
    add-iam-policy-binding \
    ambient-depth-413708 \
    --member=serviceAccount:648135870514-compute@developer.gserviceaccount.com \
    --role=roles/editor

3	git clone \
    https://github.com/GoogleCloudPlatform/hpc-toolkit.git

4	cd hpc-toolkit/

5	make

6	ghpc version v1.35.1
Built from 'main' branch.
Commit info: v1.35.1-0-gdbe05eeb

7	./ghpc create \
    examples/hpc-slurm.yaml -l \
    ERROR --vars \
    project_id=ambient-depth-413708

8	./ghpc create     examples/hpc-slurm.yaml -l     ERROR --vars     project_id=ambient-depth-413708

9	./ghpc deploy hpc-slurm

10	dr_preethibaselios@cloudshell:~/hpc-toolkit (ambient-depth-413708)$ ./ghpc deploy hpc-slurm

Ths 10 th command line gets error which is shown below.

Can you help us to fix it?
Thanks in advance

Mohan

--------------------------------------------------------------

Initializing deployment group hpc-slurm/primary
Testing if deployment group hpc-slurm/primary requires adding or changing cloud infrastructure
Deployment group hpc-slurm/primary requires adding or changing cloud infrastructure
Summary of proposed changes: Plan: 38 to add, 0 to change, 0 to destroy.
(D)isplay full proposed changes,
(A)pply proposed changes,
(S)top and exit,
(C)ontinue without applying
Please select an option [d,a,s,c]: a
Running terraform apply on deployment group hpc-slurm/primary
module.homefs.random_id.resource_name_suffix: Creating...
module.homefs.random_id.resource_name_suffix: Creation complete after 0s [id=oUDURw]
module.network.module.nat_ip_addresses["us-central1"].google_compute_address.ip[1]: Creating...
module.network.module.vpc.module.vpc.google_compute_network.network: Creating...
module.network.module.nat_ip_addresses["us-central1"].google_compute_address.ip[0]: Creating...
module.slurm_controller.module.bucket[0].google_storage_bucket.buckets["hpcslurmb2a56"]: Creating...

Error: Error creating Address: googleapi: Error 409: The resource 'projects/ambient-depth-413708/regions/us-central1/addresses/hpc-slurm-net-nat-ips-us-central1-1' already exists, alreadyExists

  with module.network.module.nat_ip_addresses["us-central1"].google_compute_address.ip[1],
  on .terraform/modules/network.nat_ip_addresses/main.tf line 50, in resource "google_compute_address" "ip":
  50: resource "google_compute_address" "ip" {

Error: Error creating Address: googleapi: Error 409: The resource 'projects/ambient-depth-413708/regions/us-central1/addresses/hpc-slurm-net-nat-ips-us-central1-0' already exists, alreadyExists

  with module.network.module.nat_ip_addresses["us-central1"].google_compute_address.ip[0],
  on .terraform/modules/network.nat_ip_addresses/main.tf line 50, in resource "google_compute_address" "ip":
  50: resource "google_compute_address" "ip" {

Error: googleapi: Error 409: Your previous request to create the named bucket succeeded and you already own it., conflict

  with module.slurm_controller.module.bucket[0].google_storage_bucket.buckets["hpcslurmb2a56"],
  on .terraform/modules/slurm_controller.bucket/main.tf line 40, in resource "google_storage_bucket" "buckets":
  40: resource "google_storage_bucket" "buckets" {

Error: Error creating Network: googleapi: Error 409: The resource 'projects/ambient-depth-413708/global/networks/hpc-slurm-net' already exists, alreadyExists

  with module.network.module.vpc.module.vpc.google_compute_network.network,
  on .terraform/modules/network.vpc/modules/vpc/main.tf line 20, in resource "google_compute_network" "network":
  20: resource "google_compute_network" "network" {

Error: exit status 1

Error: Error creating Address: googleapi: Error 409: The resource 'projects/ambient-depth-413708/regions/us-central1/addresses/hpc-slurm-net-nat-ips-us-central1-1' already exists, alreadyExists

  with module.network.module.nat_ip_addresses["us-central1"].google_compute_address.ip[1],
  on .terraform/modules/network.nat_ip_addresses/main.tf line 50, in resource "google_compute_address" "ip":
  50: resource "google_compute_address" "ip" {

Error: Error creating Address: googleapi: Error 409: The resource 'projects/ambient-depth-413708/regions/us-central1/addresses/hpc-slurm-net-nat-ips-us-central1-0' already exists, alreadyExists

  with module.network.module.nat_ip_addresses["us-central1"].google_compute_address.ip[0],
  on .terraform/modules/network.nat_ip_addresses/main.tf line 50, in resource "google_compute_address" "ip":
  50: resource "google_compute_address" "ip" {

Error: googleapi: Error 409: Your previous request to create the named bucket succeeded and you already own it., conflict

  with module.slurm_controller.module.bucket[0].google_storage_bucket.buckets["hpcslurmb2a56"],
  on .terraform/modules/slurm_controller.bucket/main.tf line 40, in resource "google_storage_bucket" "buckets":
  40: resource "google_storage_bucket" "buckets" {

Error: Error creating Network: googleapi: Error 409: The resource 'projects/ambient-depth-413708/global/networks/hpc-slurm-net' already exists, alreadyExists

  with module.network.module.vpc.module.vpc.google_compute_network.network,
  on .terraform/modules/network.vpc/modules/vpc/main.tf line 20, in resource "google_compute_network" "network":
  20: resource "google_compute_network" "network" {

**Comment by cpcpython** [comment]
It was fixed by searching: slurm , slurm-hpc and finding, closing and deleting various services, API etc. associated it (including static IPs)
Thanks
